### PR TITLE
Fix local avatar, improve speed for fetching cached photos

### PR DESCRIPTION
**Pull request by annando**
For unknown reasons some local avatars had been replaced with the default avatar. Also the functionality to fetch the avatar url is moved to a central place.

And finally in the photo cac he mechanism we now use the local file if available.

### CODE REVIEW

File: src/Module/Photo.php [hunk 1]
```diff
@@ -340,6 +342,7 @@ private static function getPhotoByid(int $id, $type, $customsize)
 					}
 				}
 				if (empty($mimetext)) {
+					Logger::info('Blubb');
```
Comment (MrPetovan) [inline src/Module/Photo.php]: Please remove debug logging.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Photo.php (hunk 1 above)
Comment (annando) [reply src/Module/Photo.php]: Örgs ...

**Review by MrPetovan — APPROVED** [review]

File: src/Module/Photo.php (hunk 1 above)
Comment (tobiasd) [reply src/Module/Photo.php]: You could create a `pre-commit` hook for your git repository in `.git/hooks/pre-commit` with a small bash script. I'm using it to `php -l` the commited PHP files with

```
#!/bin/bash

git diff --cached --name-status | while read st file; do
	# skip deleted files
	if [ "$st" == 'D' ]; then continue; fi
	# do a check only on the php files
	if [[ "$file" =~ ".php" ]] && ! /usr/bin/php -l $file; then
		echo "PHP syntax check failed for file: $file"
		exit 1
	fi
done
```

one could add a `grep` for "blubb" at the end so files with a "blubb" are not commited.

```
#!/bin/bash

git diff --cached --name-status | while read st file; do
	# skip deleted files
	if [ "$st" == 'D' ]; then continue; fi
	# do a check only on the php files
	if [[ "$file" =~ ".php" ]] && ! /usr/bin/php -l $file; then
		echo "PHP syntax check failed for file: $file"
		exit 1
	fi
	if [[ "$file" =~ ".php" ]] && /bin/grep "blubb" $file; then
		echo "The file $file contains a blubb"
		exit 1
	fi
done
```

File: src/Model/Contact.php [hunk 2]
```diff
@@ -1700,6 +1700,10 @@ private static function checkAvatarCacheByArray(array $contact, bool $no_update
 			}
 		}
 
+		if (Network::isLocalLink($contact['url']) && !empty($contact['avatar'])) {
```
Comment (tobiasd) [inline src/Model/Contact.php:1703]: The "url" field is not defined on my system for the `$contact` causing at least a PHP warning

```
PHP Warning:  Undefined array key "url" in src/Model/Contact.php on line 1703
```

When calling a DM conversation that was started before tonight's pull from the RC branch.

Comment (tobiasd) [reply src/Model/Contact.php:1703]: The friendica log has the following along side the above PHP error

```
Uncaught Exception TypeError: "Friendica\Util\Network::isLocalLink(): Argument #1 ($url) must be of type string, null given, called in /src/Model/Contact.php on line 1703" at /src/Util/Network.php line 559 
{"exception":"TypeError: Friendica\\Util\\Network::isLocalLink(): Argument #1 ($url) must be of type string, null given, called in src/Model/Contact.php on line 1703 and defined in /src/Util/Network.php:559\nStack trace:

0 /src/Model/Contact.php(1703): Friendica\\Util\\Network::isLocalLink()
1 /src/Model/Contact.php(1607): Friendica\\Model\\Contact::checkAvatarCacheByArray()
2 /src/Model/Contact.php(1653): Friendica\\Model\\Contact::getAvatarPath()
3 /mod/message.php(312): Friendica\\Model\\Contact::getThumb()
4 /src/LegacyModule.php(96): message_content()
5 /src/LegacyModule.php(73): Friendica\\LegacyModule->runModuleFunction()
6 /src/BaseModule.php(239): Friendica\\LegacyModule->content()
7 /src/App.php(711): Friendica\\BaseModule->run()\n#8 /index.php(44): Friendica\\App->runFrontend()
9 {main}"} - {"file":null,"line":null,"function":null,"uid":"1b3bd7","process_id":7457}
```

File: src/Module/Photo.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Photo.php]: Why write a script when I'm doing it anyway during my reviews?

**Pull request merged**